spextra/spextra.py

```diff
@@ -397,33 +443,39 @@ def flat_spectrum(cls, amplitude=0, waves=None):
         spex : Spextrum
             New ``Spextrum`` instance.
         """
-        if waves is None:
-            waves = _default_waves()
-        waves = _angstrom_value(waves)
-        amplitude = _abmag_qty(amplitude)
+        waves = (waves or _default_waves()).to(u.AA).value
 
         # The if-statement below also allowed amplitude.unit to be
         # u.Unit("vegamag"). Vegamag is removed from astropy, so the
         # if-statement is removed too. This assertion is added as a sanity
         # check ensure that any bugs introduced by changing the if-statement
         # is explicitly raised here, instead of silently propagated.
-        assert not amplitude.unit.to_string() == "vegamag", "The use of vegamag is deprecated."
+        assert (
+            not amplitude.unit.to_string() == "vegamag"
+        ), "The use of vegamag is deprecated."
         if amplitude.unit is u.Unit("mag"):
             spex = get_vega_spectrum()
             spex *= 10 ** (-0.4 * amplitude.value)
             # system_name = amplitude.unit
         else:
             const = ConstFlux1D(amplitude=amplitude)
-            spex = cls(modelclass=Empirical1D, points=waves,
-                       lookup_table=const(waves))
+            spex = cls(
+                modelclass=Empirical1D, points=waves, lookup_table=const(waves)
+            )
             # system_name = amplitude.unit
 
         spex.repr = f".flat_spectrum(amplitude={amplitude!r})"
         return spex
 
     @classmethod
-    def black_body_spectrum(cls, temperature=9500, amplitude=0,
-                            filter_curve=None, waves=None):
+    @u.quantity_input
+    def black_body_spectrum(
+        cls,
+        temperature: u.Quantity[u.K] = 9500 * u.K,
```

Comment:
This unfortunately broke ScopeSim_Templates, because [the temperature used there is a normal float](https://github.com/AstarVienna/ScopeSim_Templates/blob/5ebe043524a9af1d5aa6d8ef071d8ff9dc92ebf8/scopesim_templates/calibration/calibration.py#L90)

Reply:
Then we should change that there

Reply:
Yes, see https://github.com/AstarVienna/ScopeSim_Templates/pull/104 , and I added tests, because we should also not accidentally break our own products.